### PR TITLE
Allow uploading testcases for external jobs with stacktrace.

### PR DESCRIPTION
**Pull request by oliverchang**
Move record_fuzz_target from fuzz_task to data_handler (to be called
from upload) as external jobs do not run fuzz_task.

Move duplicate checks into data_handler as well.

### CODE REVIEW

File: src/python/datastore/data_handler.py [hunk 1]
```diff
@@ -1451,6 +1465,50 @@ def get_testcase_variant(testcase_id, job_type):
 # Fuzz target related functions
 # ------------------------------------------------------------------------------
 
+FUZZ_TARGET_UPDATE_FAIL_RETRIES = 5
```
Comment (oliverchang) [inline src/python/datastore/data_handler.py:1468]: unchanged. copied from fuzz_task

**Comment by oliverchang** [comment]
This needs some additional changes (marking draft for now).

**Comment by oliverchang** [comment]
Now ready for review.

File: src/appengine/handlers/upload_testcase.py [hunk 2]
```diff
@@ -380,22 +389,43 @@ def do_post(self):
...
+
+      crash_data = stack_analyzer.get_crash_data(
+          stacktrace,
+          fuzz_target=target_name,
+          symbolize_flag=False,
+          already_symbolized=True,
+          detect_ooms_and_hangs=True)
+    elif stacktrace:
+      raise helpers.EarlyExitException(
+          'Unable to specify stacktrace for non-external jobs.', 400)
```
Comment (inferno-chromium) [inline src/appengine/handlers/upload_testcase.py:412]: s/Unable/Should not specify ?

**Review by inferno-chromium — APPROVED** [review]

File: src/appengine/handlers/upload_testcase.py (hunk 2 above)
Comment (oliverchang) [reply src/appengine/handlers/upload_testcase.py]: Fixed!

**Pull request merged**